supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/GoogleLabels.scala

```diff
@@ -38,8 +41,11 @@ object GoogleLabels {
 
         val foldResult = mainText.toCharArray.foldLeft("")(appendSafe)
 
-        val startsValid = foldResult.headOption.exists(_.isLetter)
-        val endsValid = foldResult.lastOption.exists(_.isLetterOrDigit)
+        val startsValid = if (labelKey) foldResult.headOption.exists(_.isLetter) else foldResult.lastOption.exists(_.isLetterOrDigit)
+        val endsValid = (
+          foldResult.lastOption.exists(_.isLetterOrDigit)
+          || foldResult.last.equals("-") || foldResult.last.equals("_")
```

Comment:
Not sure why the first test is `lastOption` and the rest are `last`?

```suggestion
    val endsValid = foldResult.lastOption.exists(e => e.isLetterOrDigit || e == '_' || e == '-')
```

Reply:
I did not understand the difference until now